client/python/lib/ovmsclient/tfs_compat/grpc/tensors.py

```diff
@@ -14,11 +14,44 @@
 # limitations under the License.
 #
 
+from typing import NamedTuple
+from tensorflow.core.framework.tensor_shape_pb2 import TensorShapeProto
 from tensorflow.core.framework.tensor_pb2 import TensorProto
-from tensorflow.core.framework.types_pb2 import DataType
-from enum import IntEnum
+from tensorflow.core.framework.types_pb2 import DT_BOOL, DT_COMPLEX128, DT_COMPLEX64, DT_DOUBLE, DT_FLOAT, DT_HALF, DT_INT16, DT_INT32, DT_INT64, DT_INT8, DT_STRING, DT_UINT16, DT_UINT32, DT_UINT64, DT_UINT8
 import numpy as np
 
+class TensorType(NamedTuple):
+    TensorDtype: str
+    TensorProtoField: str
+
+NP_TO_TENSOR_MAP = {
+    np.float16: TensorType(TensorDtype=DT_HALF, TensorProtoField="half_val"),
+    np.float32: TensorType(TensorDtype=DT_FLOAT, TensorProtoField="float_val"),
+    np.float64: TensorType(TensorDtype=DT_DOUBLE, TensorProtoField="double_val"),
+    np.int8: TensorType(TensorDtype=DT_INT8, TensorProtoField="int_val"),
+    np.int16: TensorType(TensorDtype=DT_INT16, TensorProtoField="int_val"),
+    np.int32: TensorType(TensorDtype=DT_INT32, TensorProtoField="int_val"),
+    np.int64: TensorType(TensorDtype=DT_INT64, TensorProtoField="int64_val"),
+    np.uint8: TensorType(TensorDtype=DT_UINT8, TensorProtoField="int_val"),
+    np.uint16: TensorType(TensorDtype=DT_UINT16, TensorProtoField="int_val"),
+    np.uint32: TensorType(TensorDtype=DT_UINT32, TensorProtoField="uint32_val"),
+    np.uint64: TensorType(TensorDtype=DT_UINT64, TensorProtoField="uint64_val"),
+    np.complex64: TensorType(TensorDtype=DT_COMPLEX64, TensorProtoField="scomplex_val"),
+    np.complex128: TensorType(TensorDtype=DT_COMPLEX128, TensorProtoField="dcomplex_val"),
+    np.str: TensorType(TensorDtype=DT_STRING, TensorProtoField="string_val"),
+    np.bool: TensorType(TensorDtype=DT_BOOL, TensorProtoField="bool_val"),
+}
+
+TENSOR_TO_NP_MAP = {v.TensorDtype: k for k, v in NP_TO_TENSOR_MAP.items()}
+
+TENSOR_DTYPE_TO_PROTOFIELD = {v.TensorDtype : v.TensorProtoField for v in NP_TO_TENSOR_MAP.values()}
+
+def _is_array_like(value):
+    return isinstance(value, (list, np.ndarray))
+
+def _is_shape_valid(shape):
```

Comment:
Perhaps we should make sure that values inside shape list or tuple are unsigned integers?